packages/api/routes/apikey/createKeyMiddleware.js

```diff
@@ -0,0 +1,12 @@
+const db_apikey = require("../../lib/db.apikey");
+
+// Return a function for creating an API key for a given user.
+module.exports = function createKeyMiddleware() {
+  return async (req, res, next) => {
+    const apikey = await db_apikey.attach();
+    apikey.createKey(req.body.user).then(doc => {
```

Comment:
Error conditions this needs to handle:

 - request body is empty
 - request body has no `user` property
 - `user` is empty string
 - database connection fails

And maybe others.  Use your imagination. :)

Reply:
I thought more about the error conditions this afternoon and I think I was too hasty.  There are other parts of SPAship that _dont_ have super robust error checking, so your code would fit right in.  It's okay if you want to wait until after MVP to add really thorough handling of error conditions.  I'll leave it up to you @sayak-sarkar.  (I'm leaving this same message on my other error condition comment.)